src/reanimated2/animation/springUtils.ts

```diff
@@ -154,6 +162,57 @@ export function initialCalculations(
   }
 }
 
+/** We make an assumption that we can manipulate zeta without changing duration of movement.
+ *  According to theory this change is small and tests shows that we can indeed ignore it.
+ */
+export function scaleZetaToMatchClamps(
+  animation: SpringAnimation,
+  clamp: [number, number]
+) {
+  'worklet';
+  const { zeta, toValue, startValue } = animation;
+  if (Number(toValue) === startValue) {
```

Comment:
Why do we convert `toValue` into a number here? Does it mean that we no longer support colors?

Reply:
I've just tested and it looks that color handling is a bit buggy with our main code 😱 
Something like
```javascript
 const style = useAnimatedStyle(() => {
    return {
      backgroundColor: withSpring(randomColor.value, config),
    };
  });
```
where randomColor is a colorname like 'red' or 'blue' doesn't always run.

Anyway toValue is expected to be a number, since in `src/reanimated2/animation/util.ts` we have a lot of functions like `colorOnStart` and `numberOnStart` that are expected to convert input to numbers and then join it again.

Reply:
Also if `toValue` weren't `number` it would be difficult wo handle it, since value is always a number.

Reply:
Btw. logic inside `src/reanimated2/animation/util.ts` is very important and complicated, I think I'd suggest splitting it into multiple files, WDYT?